modules/apps/wiki/wiki-engine-input-editor-common/src/main/java/com/liferay/wiki/engine/input/editor/common/internal/util/WikiEngineInputEditorCommonComponentProvider.java

```diff
@@ -34,25 +30,14 @@ public class WikiEngineInputEditorCommonComponentProvider {
 	}
 
 	public ServletContext getServletContext() {
-		return _servletContext;
-	}
-
-	@Activate
-	protected void activate() {
-		_wikiEngineInputEditorCommonComponentProvider = this;
-	}
-
-	@Deactivate
-	protected void deactivate() {
-		_wikiEngineInputEditorCommonComponentProvider = null;
+		return _servletContextSnapshot.get();
 	}
 
+	private static final Snapshot<ServletContext> _servletContextSnapshot =
+		new Snapshot<>(
+			WikiEngineInputEditorCommonComponentProvider.class,
+			ServletContext.class);
 	private static WikiEngineInputEditorCommonComponentProvider
 		_wikiEngineInputEditorCommonComponentProvider;
 
```

Comment:
Hey @RenanUmV, now the field `_wikiEngineInputEditorCommonComponentProvider` is never initialized. By the way, you do not need this more. 

Thx